hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -477,14 +479,19 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
     }
 
     boolean shouldCombine = cfg.filterDupes || cfg.operation.equals(WriteOperationType.UPSERT);
+    boolean isDropPartitionColumns = props.getBoolean(DROP_PARTITION_COLUMNS.key());
+    String partitionColumns = HoodieSparkUtils.getPartitionColumns(keyGenerator, props);
+    List<String> listOfPartitionColumns = Arrays.asList(partitionColumns.split(","));
     JavaRDD<GenericRecord> avroRDD = avroRDDOptional.get();
+    HoodieKey key = keyGenerator.getKey(avroRDD.first());
```

Comment:
not sure if this is right. we have to move this within map (). for each genRec, we have to first run through key generator and then modify the genRec if need be (drop partition cols). 
Also, can we move lines 482 to 494 to separate method and keep this method cleaner/leaner.

Reply:
To make it cleaner I picked the first record for generating the key, I see your point, updated it.